lib/mail/Substitution.php

```diff
@@ -69,15 +69,15 @@ public function getKey()
     /**
      * Add the value on a Substitution object
      *
-     * @param string|array|bool|int $value Substitution value
+     * @param string|array|object|bool|int|float $value Substitution value
      *
      * @throws \SendGrid\Mail\TypeException
      * @return null
      */
     public function setValue($value)
     {
-        if (!is_string($value) && !is_array($value) && !is_object($value) &&!is_bool($value) &&!is_int($value)) {
-            throw new TypeException('$value must be of type string, array or object.');
+        if (!is_string($value) && !is_array($value) && !is_object($value) && !is_bool($value) && !is_numeric($value)) {
+            throw new TypeException("The value: '$value' must be of type string, array, object, bool, int or float.");
```

Comment:
Drop the colon: `"The value '$value' ..`

And needs [oxford comma](https://www.lexico.com/definition/oxford_comma).